sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -30,50 +61,60 @@ export class ApplicationOfferingChangeRequestService {
    */
   async getEligibleApplications(
     locationId: number,
-    paginationOptions: PaginationOptions,
-  ): Promise<PaginatedResults<Application>> {
+    options: {
+      pagination?: PaginationOptions;
+      applicationId?: number;
+    },
+  ): Promise<PaginatedResults<Application> | Application> {
     const applicationQuery = this.applicationRepo
       .createQueryBuilder("application")
       .select([
         "application.applicationNumber",
         "application.id",
+        "programYear.id",
         "currentAssessment.id",
+        "educationProgram.id",
+        "offering.id",
         "offering.studyStartDate",
         "offering.studyEndDate",
+        "offering.offeringIntensity",
         "student.id",
         "user.firstName",
         "user.lastName",
       ])
+      .innerJoin("application.programYear", "programYear")
       .innerJoin("application.currentAssessment", "currentAssessment")
       .innerJoin("currentAssessment.offering", "offering")
+      .innerJoin("offering.educationProgram", "educationProgram")
       .leftJoin(
         "application.applicationOfferingChangeRequest",
         "applicationOfferingChangeRequest",
+        "applicationOfferingChangeRequest.applicationOfferingChangeRequestStatus IN (:...status)",
+        {
+          status: [
+            ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+            ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
+          ],
+        },
       )
       .innerJoin("application.student", "student")
       .innerJoin("student.user", "user")
-      .where(
-        new Brackets((qb) =>
-          qb
-            .where("applicationOfferingChangeRequest.id IS NULL")
-            .orWhere(
-              "applicationOfferingChangeRequest.applicationOfferingChangeRequestStatus NOT IN (:...status)",
-              {
-                status: [
-                  ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
-                  ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
-                ],
-              },
-            ),
-        ),
-      )
+      .where("applicationOfferingChangeRequest.id IS NULL")
       .andWhere("application.location.id = :locationId", { locationId })
       .andWhere("application.applicationStatus = :applicationStatus", {
         applicationStatus: ApplicationStatus.Completed,
       })
       .andWhere("application.isArchived = false");
 
-    if (paginationOptions.searchCriteria?.trim()) {
+    if (options.applicationId) {
+      applicationQuery.andWhere("application.id = :applicationId", {
+        applicationId: options.applicationId,
+      });
+      // If searching by application id there is no need for pagination.
+      return applicationQuery.getOne();
+    }
+
+    if (options.pagination.searchCriteria?.trim()) {
```

Comment:
shouldn't we add null check to `pagination`  `options.pagination?.searchCriteria`

Reply:
This method has 2 possible ways to be called and it will ensure that either the `options.applicationId` will be provided or `options.pagination`. At this stage it was already determined that `options.applicationId` was not provided which means that `options.pagination` must be available.